scripting/src/main/java/org/infinispan/scripting/impl/DistributedRunner.java

```diff
@@ -34,13 +35,12 @@ private DistributedRunner() {
       DefaultExecutorService des = new DefaultExecutorService(masterCacheNode);
       try {
          List<Future<T>> tasks = des.submitEverywhere(new DistributedScript<T>(metadata));
-         List<NotifyingFuture<T>> notifyingTasks = new ArrayList<>(tasks.size());
-         for (Future<T> task : tasks) {
-               notifyingTasks.add((NotifyingFuture<T>) task);
+         List<CompletableFuture<T>> all = new ArrayList<>(tasks.size());
```

Comment:
You could do this more succinctly with `tasks.stream().map(CompletableFutures::connect)`. Not sure if it'll work with `::` since you seem to need the cast...

Also, your CompletableFutures.connect, or asCompletableFuture, could take Future instead of NotifyingFuture and then users wouldn't need to do casting. Within the converter method you'd check the real type...